tests/Microsoft.Identity.Test.Integration.net45/Infrastructure/HttpTelemetryRecorder.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Identity.Client.TelemetryCore;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Integration.net45.Infrastructure
+{
+    public class HttpTelemetryRecorder
```

Comment:
This is looking much better. I would add some methods that make assertions, smth like 

`AssertCurrentTelemetry(expectedApiId, expectedErrorCode ...)`